ui/pages/bridge/token-insights-modal/token-insights-modal.tsx

```diff
@@ -0,0 +1,268 @@
+  token: TokenInsightsToken | null;
+};
+
+type MarketDataRowProps = {
+  label: string;
+  value: React.ReactNode;
+  'data-testid'?: string;
+};
+
+const MarketDataRow: React.FC<MarketDataRowProps> = ({
+  label,
+  value,
+  'data-testid': dataTestId,
+}) => (
+  <Box marginBottom={3} data-testid={dataTestId}>
+    <Text variant={TextVariant.BodyMd} color={TextColor.TextAlternative}>
+      {label}
+    </Text>
+    <Box alignItems={BoxAlignItems.Center}>{value}</Box>
+  </Box>
+);
+
+export const TokenInsightsModal: React.FC<TokenInsightsModalProps> = ({
+  isOpen,
+  onClose,
+  token,
+}) => {
+  const t = useI18nContext();
+  const trackEvent = React.useContext(MetaMetricsContext);
+  const dialogRef = React.useRef<HTMLElement | null>(null);
+
+  const { marketData, marketDataFiat } = useTokenInsightsData(token);
+
+  useEffect(() => {
+    if (isOpen && token) {
+      trackEvent({
+        event: 'Token Insights Modal Opened',
+        category: MetaMetricsEventCategory.Swaps,
+        properties: {
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          token_symbol: token.symbol,
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          token_address: token.address,
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          chain_id: token.chainId,
+        },
+      });
+    }
+  }, [isOpen, token, trackEvent]);
+
+  // Ensure only the top modal closes on outside click by intercepting
+  // document mousedown and stopping propagation.
+  useEffect(() => {
+    if (!isOpen) {
+      return () => undefined;
+    }
+    const handleDocMouseDownCapture = (event: MouseEvent) => {
+      const el = dialogRef.current;
+      // this prevents the modal from closing when clicking on the modal content
+      if (el && !el.contains(event.target as Node)) {
+        onClose();
+        event.stopPropagation();
+        event.stopImmediatePropagation();
+        event.preventDefault();
+      }
+    };
+    document.addEventListener('mousedown', handleDocMouseDownCapture, true);
+    return () => {
+      document.removeEventListener(
+        'mousedown',
+        handleDocMouseDownCapture,
+        true,
+      );
+    };
+  }, [isOpen, onClose]);
+
+  const priceChange24h = marketData?.pricePercentChange1d || 0;
+
+  const handleCopyAddress = useCallback(() => {
+    if (token) {
+      trackEvent({
+        event: 'Token Contract Address Copied',
+        category: MetaMetricsEventCategory.Swaps,
+        properties: {
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          token_symbol: token.symbol,
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          token_address: token.address,
+        },
+      });
+    }
+  }, [token, trackEvent]);
+
+  if (!token) {
+    return null;
+  }
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} className="token-insights-modal">
```

Comment:
I don't know why you would need to override the width, but if it's absolutely needed you can put here directly without needing a brand new SCSS file (we're trying to move away from these) and use more Tailwind

`className="max-w-[400px]"`